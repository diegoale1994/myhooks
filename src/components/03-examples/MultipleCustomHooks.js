import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import '../02-useEffect/effects.css'
import { useCounter } from '../../hooks/useCounter'
export const MultipleCustomHooks = () => {

    const { state, increment } = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${state}`);

    const { author, quote } = !!data && data[0];

    return (
        <div>
            <h1>Braking bad quotes</h1>
            <hr />
            {loading ?
                (
                    <div className="alert alert-info text-center">Loading</div>
                )
                :
                (
                    <blockquote className="blockquote text-right">
                        <p className="mb-0">{quote}</p>
                        <footer className="blockquote-footer">{author}</footer>
                    </blockquote>
                )}
            {!loading && (
                <button className="btn btn-primary" onClick={increment}>Next phrase</button>
            )}
    <br />
    <br />
    <br />
        </div>
    )
}
