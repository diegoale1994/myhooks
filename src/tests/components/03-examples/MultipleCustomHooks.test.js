import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('pruebas en MultipleCustomHooks', () => {

    useCounter.mockReturnValue({
        state: 10,
        increment: () => {}
    })

    test('debe mostrarse correctamente', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar la informacion', () => {
        useFetch.mockReturnValue({
            data: [{
                author:'Diego',
                quote: 'Hello world'
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hello world');
        expect(wrapper.find('footer').text().trim()).toBe('Diego');
        console.log(wrapper.html())
    })
    
    
})
