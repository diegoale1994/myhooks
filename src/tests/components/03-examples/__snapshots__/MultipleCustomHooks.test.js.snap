// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en MultipleCustomHooks debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Braking bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading
  </div>
  <br />
  <br />
  <br />
</div>
`;
